c7n/resources/apigw.py

```diff
@@ -119,6 +119,45 @@ class resource_type(object):
         dimension = 'GatewayName'
 
 
+@RestAPI.action_registry.register('update')
+class UpdateAPI(BaseAction):
+    """Update configuration of a REST API.
+
+    Non-exhaustive list of updateable attributes.
+    https://docs.aws.amazon.com/apigateway/api-reference/link-relation/restapi-update/#remarks
+
+    :example:
+
+    contrived example to update description on api gateways
+
+    .. code-block:: yaml
+
+       policies:
+         - name: apigw-description
+           filters:
+             - description: empty
+           actions:
+             - type: update
+               patch:
+                - op: replace
+                  path: /description
+                  value: "not empty :-)"
+    """
+    permissions = ('apigateway:PATCH',)
+    schema = utils.type_schema(
+        'update',
+        patch={'type': 'array', 'items': OP_SCHEMA},
+        required=['patch'])
+
+    def process(self, resources):
+        client = utils.local_session(
+            self.manager.session_factory).client('apigateway')
+        for r in resources:
+            client.update_rest_api(
```

Comment:
Potentially attempt retry on errors that can be retried as it could break on one resource and that prevent remaining resources from being updated. Most errors seem to not warrant a retry though. 

https://docs.aws.amazon.com/apigateway/api-reference/handling-errors/

Reply:
retry on updates is always a bit sketchy in distributed systems without a conditional check against the source version/hash that the change is being made against. that said i'm more concerned with the set cardinality of the resources with the rate of change allowed by the control plane api, it looks like the defaults there can be 1k+ so it does seem to warrant some retry logic to make resource set progress. done.